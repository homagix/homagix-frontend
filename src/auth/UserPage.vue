<script setup lang="ts">
import { useRouter } from "vue-router"
import AppButton from "../components/AppButton.vue"

const router = useRouter()

function gotoRecipes() {
  router.push("/recipes")
}
</script>

<template>
  <h2 class="title is-4">Dein Zugang</h2>
  <p>
    Speichere diese Seite als Bookmark oder Favorit, dann kommst du einfach wieder hierher zurück, ohne dich einloggen
    zu müssen.
  </p>
  <p>Da jede:r mit dem Bookmark hierher kommt, solltest du diesen Link nicht an andere weiter geben.</p>

  <div class="button-list">
    <AppButton id="recipes-button" @click="gotoRecipes">Zu den Rezepten</AppButton>
  </div>
</template>
