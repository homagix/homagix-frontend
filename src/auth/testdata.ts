import { vi } from "vitest"
import { createStore } from "vuex"
import { mount } from "@vue/test-utils"
import { createRouterMock } from "../mocks/router"
import type { User } from "."

export const token =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjMiLCJmaXJzdE5hbWUiOiJhYmMiLCJpYXQiOjE2NDY5Nzg0NTUsImV4cCI6MTY0NzA2NDg1NX0.kClNCHxRnm_rvNpkdwdHgDsVebJfOzZ66Z_5epxkKNU"

export const user: User = { id: "123", name: "abc", accessCode: "456" }

export function mountComponent<C>(component: C) {
  const router = createRouterMock(component, ["/my", "/register"])
  const registerUserAction = vi.fn()
  const store = createStore({
    state() {
      return {}
    },
    actions: {
      REGISTER_USER: registerUserAction,
    },
  })
  const wrapper = mount(component, { global: { plugins: [store, router] } })
  return { wrapper, registerUserAction, router }
}
