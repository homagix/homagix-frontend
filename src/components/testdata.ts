import { Dish, Ingredient } from "@/types"

export const avocado = { id: "42", name: "Avocado", unit: "Stk" } as Ingredient
export const tomato = { id: "43", name: "Tomate", unit: "Stk" } as Ingredient

export const guacamole: Dish = {
  id: "5",
  name: "Guacamole",
  items: [
    { id: avocado.id, amount: 2 },
    { id: tomato.id, amount: 2 },
  ],
  alwaysOnList: false,
  images: ["guacamole.jpg"],
  isFavorite: false,
  isEditable: true,
}

export const tomatoSoup: Dish = {
  id: "7",
  name: "Tomatensuppe",
  items: [{ id: tomato.id, amount: 4 }],
  alwaysOnList: false,
  images: ["tomatoSoup.jpg"],
  isFavorite: false,
  isEditable: false,
}
