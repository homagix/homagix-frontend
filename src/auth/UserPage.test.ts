import { mount } from "@vue/test-utils"
import { describe, expect, it, vi } from "vitest"
import UserPage from "@/auth/UserPage.vue"
import { mountComponent } from "./testdata"

describe("UserPage", () => {
  it("should render", () => {
    const { wrapper } = mountComponent(UserPage)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should go to recipies page if button is clicked", async () => {
    const { wrapper, router } = mountComponent(UserPage)
    router.push = vi.fn()
    await wrapper.find("#recipes-button").trigger("click")
    expect(router.push).toHaveBeenCalledWith("/recipes")
  })
})
