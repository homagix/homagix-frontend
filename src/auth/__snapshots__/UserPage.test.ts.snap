// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserPage > should render 1`] = `
"<h2 class=\\"title is-4\\">Dein Zugang</h2>
<p> Speichere diese Seite als Bookmark oder Favorit, dann kommst du einfach wieder hierher zurück, ohne dich einloggen zu müssen. </p>
<p>Da jede:r mit dem Bookmark hierher kommt, solltest du diesen Link nicht an andere weiter geben.</p>
<div class=\\"button-list\\">
  <app-button-stub id=\\"recipes-button\\"></app-button-stub>
</div>"
`;
